### PR TITLE
(BIDS 3226) API: Block Details v2 (Internal)

### PR DESCRIPTION
**Pull request by remoterami**

### CODE REVIEW

File: backend/pkg/api/handlers/common.go [hunk 1]
```diff
@@ -359,6 +359,31 @@ func (h *HandlerService) getDashboardPremiumPerks(ctx context.Context, id types.
...
+		return 0, 0, err
+	}
+	chainId := v.checkNetwork(req.Network)
+	block, err := h.dai.GetLatestBlock()
+	if err != nil {
+		return 0, 0, err
+	}
+	if req.Block != "latest" {
+		block = v.checkUintMinMax(req.Block, 0, block, "block")
+	}
```
Comment (LuccaBitfly) [inline backend/pkg/api/handlers/common.go:379]: You should only fetch the latest block if the user requests latest.

**Review by LuccaBitfly — COMMENTED** [review]
lgtm so far, pls make one of everything for slots too 😅

**Review by LuccaBitfly — APPROVED** [review]
lgtm, don't forget to let FE team approve structs

File: frontend/types/api/block.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  gas_used: number /* uint64 */;
+  gas_price?: string /* decimal.Decimal */;
+  excess_gas: number /* uint64 */;
+  burned_fees?: string /* decimal.Decimal */;
+}
+export interface BlockOverview {
+  /**
+   * General
+   */
+  block: number /* uint64 */;
```
Comment (MauserBitfly) [inline frontend/types/api/block.ts:76]: Some properties overlap between the `BlockOverview` and the `BlockExecutionPayload`. 
Wouldn't it be enough to have them just in the  `BlockExecutionPayload` ?

**Review by MauserBitfly — COMMENTED** [review]
Looks good from the FE point of view, just have one remark

**Review by MauserBitfly — APPROVED** [review]
LGTM

**Pull request merged**